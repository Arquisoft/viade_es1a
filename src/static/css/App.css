.app {
  width: 90%;
  height: 90%;
  background-color:rgb(255, 255, 255) ;
}

.nav{

  position: relative;
  width: 100%;
  height: 100%;
  background-color:rgb(248, 144, 144) ;
  top:-10%;
  bottom:0;
  right:0;
  left:0;
  
}


/* .usrCmp{

} */
.user{
  position: relative;
  top:0%;
  bottom:0;
  right:0;
  left:10%;
}
/* .notCmp{

} */


.Campanita-ico{
  width:3%;
  min-width: 20px;
  max-width: 50px; 
}


.notification{
  position: relative;
  top:-85%;
  bottom:70%;
  right:0;
  left:35%;
}

.btnNtf{
    
  position: relative;
  top:3%;
  bottom:0;
  right:0;
  left:5%;
  width: 18%;
  height: 5%;
 
  color: #565656;
 
  border: solid 2px #a4fa1a;
 
  border-radius: 4px;
 
  background: #fff;

  cursor: pointer;
}
/* .lgnBtn{

} */

.logout{

  position: relative;
  top:-20;
  bottom:10%;
  right:0;
  left:85.3%;
  width: 0;
  height: 1%;
 
  min-width: 280px;
  /*max-width: 400px;*/
 
  color: #565656;
 
  padding: 8px;
 
  font-size: 14px;
 
  font-weight: bold;
 
  border: solid 2px #a4fa1a;
 
  border-radius: 4px;
 
  background: #fff;
 
  cursor: pointer;
}

.map{
  height: 50%;
}

.listaRuta{
  position: relative;
  top:0%;
  bottom:0;
  right:40%;
  left:-70%;
  width: 60%;
}
.share{
  text-align: center;
  width: 40%;
  font-size: 20px;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(0deg);
  }
}
