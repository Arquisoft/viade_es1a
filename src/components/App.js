import React from 'react';
import logo from '../assets/images/ViaDe.svg';
import '../assets/css/login.css';
import UserStore from './login/UserStore';
import LoginForm from './login/LoginForm';
import SubmitButton from './login/SubmitButton';
import { observer } from 'mobx-react';

function Imagen() {
  return (<img src={logo} className="App-logo" alt="logo" />);
}

class App extends React.Component {

  async componentDidMount() {
    try {
      let res = await fetch('./isLoggedIn', {
        method: 'post',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json'
        }
      });
      let result = await res.json();
      if (result && result.sucess) {
        UserStore.loading = false;
        UserStore.isLoggedIn = true;
        UserStore.username = result.username;
      }
      else {
        UserStore.loading = false;
        UserStore.isLoggedIn = false;
      }
    }
    catch (e) {
      UserStore.loading = false;
      UserStore.isLoggedIn = false;
    }
  }

  async doLogout() {
    try {
      let res = await fetch('./logout', {
        method: 'post',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json'
        }
      });
      let result = await res.json();
      if (result && result.sucess) {
        UserStore.isLoggedIn = false;
        UserStore.username = '';
      }
    }
    catch (e) {
      console.log(e);
    }
  }
  render() {
    if (UserStore.loading) {
      return (
        <div className="App">
          <div className="container">
            Cargando, espere...
          </div>
        </div>
      );
    }
    else {
      if (UserStore.isLoggedIn) {
        return (
          <div className="App">
            <div className="container">
              <header className="App-header">
                <img src={logo} className="App-logo" alt="logo" />
              </header>
              Bienvenido {UserStore.username}

              <SubmitButton
                text={'Desconectar'}
                desabled={false}
                onClick={() => this.doLogout}
              />
            </div>
          </div>
        );
      }
      return (
        <div className="App">
          <div className="container">
            {Imagen()}
            <LoginForm />
          </div>
        </div>
      );
    }
  }
}

export default observer(App);
