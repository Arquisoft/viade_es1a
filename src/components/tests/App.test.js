import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import App from '../App';

test('renders proyecto en github link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/En desarollo/i);
  expect(linkElement).toBeInTheDocument();
});
