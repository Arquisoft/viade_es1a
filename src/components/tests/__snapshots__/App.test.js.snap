// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loginButton 1`] = `
<button
  className="solid auth login"
  onClick={[Function]}
>
  Log in
</button>
`;

exports[`main 1`] = `
<div
  className="App"
>
  <div
    className="container"
  />
</div>
`;

exports[`uploadToPod 1`] = `<uploadToPod />`;

exports[`user 1`] = `
<section>
  <div
    className="col-sm"
  >
    <span>
      Estas logueado como: 
      <a
        href={null}
      >
        <span
          className="solid value pending"
        />
      </a>
    </span>
  </div>
</section>
`;
